src/ru/mangabuff/src/eu/kanade/tachiyomi/extension/ru/mangabuff/MangaBuff.kt

```diff
@@ -46,14 +48,16 @@ class MangaBuff : ParsedHttpSource() {
 
         if (request.method == "POST" && request.header("X-CSRF-TOKEN") == null) {
             val newRequest = request.newBuilder()
+                .addHeader("X-Requested-With", "XMLHttpRequest")
+
             val token = getToken()
             val response = chain.proceed(
                 newRequest
                     .addHeader("X-CSRF-TOKEN", token)
                     .build(),
             )
 
-            if (response.code == 419) {
+            if (!response.isSuccessful && response.code == 419) {
```

Comment:
This is essentially equivalent to:
```kt
if (response.code !in 200..299 && response.code == 419) {
    // ...
}
```
If the second condition is true then the first condition is always true.

https://github.com/square/okhttp/blob/59cbf64f6ba98e2c8f95bf9db41dc47ad2232f94/okhttp/src/commonJvmAndroid/kotlin/okhttp3/Response.kt#L155

https://github.com/square/okhttp/blob/59cbf64f6ba98e2c8f95bf9db41dc47ad2232f94/okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/-ResponseCommon.kt#L79-L80